akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -44,6 +44,16 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
 
       def render(h: HttpHeader) = r ~~ h ~~ CrLf
 
+      val hostString: InetSocketAddress ⇒ String = {
+        // Retrieve the original host string that was given (IP or DNS name) if running Java 7 or later
+        // using the getHostString() method. This avoids a reverse DNS query from calling getHostName()
+        // if the original host string is an IP address.
+        val clazz = classOf[InetSocketAddress]
+        address ⇒ clazz.getMethods.collectFirst {
+          case m if m.getName == "getHostString" => m.invoke(address)
+        }.getOrElse(address.getHostName).asInstanceOf[String]
+      }
```

Comment:
> It's primarily there to make sure we get the right getHostString method


Ok, just validated that ```m.invoke(address)``` is checking for number of parameters in method so require(m.getParameterCount == 0) is redundant from this standpoint. However, return type is not validating by ```invoke``` so we do need fist require call to fall back to ```getHostName```